#159965296 Configure App for Heroku

pngg opened the pull request:
#### What does this PR do?
Configures App for deployment on Heroku

#### Description of Task to be completed
- Setup app for Heroku deployment
- Setup ```staging``` and ```production``` environments on Heroku

#### How should this be manually tested?
- Navigate to the following URLs
[`staging environment`](https://ah-magnificent6-staging.herokuapp.com/)
[`production environment`](https://ah-magnificent6.herokuapp.com/)

#### What are the relevant pivotal tracker stories?
[Setup and deploy base application to Heroku](https://www.pivotaltracker.com/story/show/159965296)

Commit:
Remove .DS_Store

Commit:
[Chore #159965296] Remove .vscode

Commit:
[Chore #159965296] Update requirements.txt

Commit:
[Chore #159965296] Add Procfile

Commit:
[Chore #159965296] Update requirements.txt

Commit:
feat(deploy): configure app for Heroku

- implement configuration settings to setup app for Heroku deployment

meshack-mbuvi commented on .gitignore:
```diff
@@ -91,3 +91,6 @@ ENV/
 
 # SQLite3
 db.sqlite3
+
```
These files are not needed. Please confirm.

meshack-mbuvi commented on .gitignore:
```diff
@@ -91,3 +91,6 @@ ENV/
 db.sqlite3
+
+.DS_Store
+.vscode
```
Remove these files.

parseen254 commented on authors/settings.py:
```diff
@@ -11,6 +11,7 @@
 """
 
 import os
+import django_heroku
```
Avoid use of preset libraries, for your own learning well being.

pngg replied:
Removed django_heroku package and now reconfigured for manual deploy.

Commit:
Add runtime.txt and move Procfile to project root

Commit:
Update requirements.txt

Commit:
chore(deploy): setup and deploy base application to Heroku

- configure app for hosting on Heroku
- remove Django-Heroku package
- add runtime.txt
- update requirements.txt
- move Procfile to project root

[Finishes #159965296]

Commit:
[Chore #159965296] Update Procfile

mikenthiwa committed:
[Chore #159965296] Deploy application automatically via travis

meshack-mbuvi commented:
Resolve conflicts.

Commit:
[Chore #159965296] Configure allowed hosts

parseen254 commented on requirements.txt:
```diff
@@ -1,8 +1,19 @@
+psycopg2==2.7.5
+psycopg2-binary==2.7.5
 PyJWT==1.4.2
+pylint==2.1.1
```
@pngg please remove non-essential requirements eg pylint.

mikenthiwa replied:
@parseendavid we have removed the unnecessary packages

mikenthiwa committed:
[Chore #159965296] Remove unnecessary third party packages from requirements
